mesonbuild/mesonlib.py

```diff
@@ -516,7 +514,7 @@ def __init__(self, s):
         # non-alphanumeric separators are discarded
         sequences = [m for m in sequences if not re.match(r'[^a-zA-Z\d]+', m.group(1))]
         # numeric sequences have leading zeroes discarded
-        sequences = [re.sub(r'^0+(\d)', r'\1', m.group(1), 1) for m in sequences]
+        sequences = [int(m.group(1)) if m.group(1).isdigit() else m.group(1) for m in sequences]
```

Comment:
I think this won't handle numeric character sequences corresponding to numbers larger than MAXINT correctly.  Possibly not a practical concern... :)

Reply:
It shouldn't actually be a problem, if we create an int so big that python 3 cant represent it, we'd have run out of memory.

Reply:
Oh yeah, sys.maxint isn't a thing on python3.